### DISCUSSION TITLE
PIO PWM switching for tone generation

### DISCUSSION THREAD

**Discussion by martinerk0** [earlephilhower/arduino-pico#1115] [General]
Hi, how would one implement something like https://github.com/teckel12/arduino-toneac on arduino-pico?

I guess it needs to sync PWM generation on both pins somehow using PIO of Pico.

**Comment by earlephilhower** [comment]
`Tone` is done by the PIO, not PWM, so there's nothing related to PWM here .  If you want to add an inverted output (but I fail to see how this helps things in any way) you'll need to add a new sideset pin and adjust accordingly.  Check out `tone2.pio` in the `cores/rp2040` dir.
